.github/workflows/build-test.yml

```diff
@@ -0,0 +1,57 @@
+name: Check
+
+on:
+  pull_request:
+
+  push:
+    branches:
+      - main # Check main branch after merge
+
+concurrency:
+  # Only run once for latest commit per ref and cancel other (previous) runs.
+  group: ci-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  check-code-style:
+    name: Code Style
+    uses: playframework/.github/.github/workflows/sbt.yml@v1
+    with:
+      cmd: sbt scalafmtCheckAll scalafmtSbtCheck
+
+#  MiMa is not enabled yet, since this library is marked as "API may change".
+#  check-binary-compatibility:
+#    name: Binary Compatibility
+#    uses: playframework/.github/.github/workflows/binary-check.yml@v1
+
+  check-docs:
+    name: Docs
+    uses: playframework/.github/.github/workflows/sbt.yml@v1
+    with:
+      cmd: sbt paradox
+
+  check-headers:
+    name: Headers
+    uses: playframework/.github/.github/workflows/sbt.yml@v1
+    with:
+      cmd: sbt headerCheckAll
```

Comment:
It's not really important, but in all other GHA workflow we have so far `headerCheckAll`resides in the `check-code-style`job. No need to have a dedicted job for that.

Reply:
Done.